packages/zoe/src/cleanProposal.js

```diff
@@ -88,6 +94,57 @@ export const cleanKeywords = keywordRecord => {
   return keywords;
 };
 
+const assertAfterDeadlineExitRule = exit => {
+  const expectedAfterDeadlineKeys = ['timer', 'deadline'];
+  assertKeysAllowed(expectedAfterDeadlineKeys, exit.afterDeadline);
+  assert(exit.afterDeadline.timer !== undefined, X`timer must be defined`);
+  assert(
+    typeof exit.afterDeadline.deadline === 'bigint' &&
+      isNat(exit.afterDeadline.deadline),
+    X`deadline must be a Nat BigInt`,
+  );
+  // timers must have a 'setWakeup' function which takes a deadline
+  // and an object as arguments.
+  // TODO: how to check methods on presences?
```

Comment:
The presence doesn't have methods. It is a presence of a remote object that has methods. The remote object is assumed to be on an untrusted machine so there is little security benefit in checking these alleged methods anyway. It could be a useful diagnostic for bugs, but not more so than any other time we get a reference to a remote object we expect to honor a particular API.

```suggestion
```